File: ios/RNContactPickerViewManager.m
#import <ContactsUI/ContactsUI.h>

#import <React/RCTComponent.h>
#import <React/RCTViewManager.h>


@interface RCT_EXTERN_MODULE(RNContactPickerViewManager, RCTViewManager)

RCT_EXTERN_METHOD((UIView *)view);

RCT_EXPORT_VIEW_PROPERTY(onCancel, RCTBubblingEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onSelectContact, RCTBubblingEventBlock)

@end

File: ios/RNContactPickerViewManager.swift
import Foundation

@objc(RNContactPickerViewManager)
class RNContactPickerViewManager: RCTViewManager {
  @objc override func constantsToExport() -> [String: Any] {
    return [
      "dark": "dark",
      "light": "light",
      "iconOnly": "iconOnly",
      "standard": "standard",
      "wide": "wide",
    ]
  }
  
  @objc override func view() -> UIView {
    return RNContactPickerView()
  }
}
